src/Stripe.net/Services/Events/EventUtility.cs

```diff
@@ -156,12 +156,14 @@ public static void ValidateSignature(string json, string stripeSignatureHeader,
                     "Make sure you're using the correct webhook secret (whsec_) and confirm the incoming request came from Stripe.");
             }
 
-            var webhookUtc = Convert.ToInt32(signatureItems["t"].FirstOrDefault());
-
-            if (Math.Abs(utcNow - webhookUtc) > tolerance)
+            if (tolerance != null)
             {
-                throw new StripeException(
-                    "The webhook cannot be processed because the current timestamp is outside of the allowed tolerance.");
+                var webhookUtc = Convert.ToInt32(signatureItems["t"].FirstOrDefault());
```

Comment:
```suggestion
                var webhookUtc = Convert.ToInt64(signatureItems["t"].FirstOrDefault());
```

To convert this type to a ``long`` for the math that comes right afterwards.